### DISCUSSION TITLE
Model prediction: I can't find a way (within the library) to get the name of the topic instead of the number

### DISCUSSION THREAD

**Discussion by Aymen-lab** [MaartenGr/BERTopic#1075] [Ideas]
Hi, 

I didn't find a way to extract the topic name after customzing it.
topic_model.set_topic_labels({0: "Healthcare", 1: "Politics", 2:"Economy"})
topic_model.custom_labels_

Weather I'm dealing with train or test data.
**For train** this is basically the dedicated method: topic_model.topics_[:]
**For test** this should be the approach: 
test_data = pd.DataFrame({'Sentences': ['The cat in the hat.', 'The dog on the log.', 'The fox in the box.']})
topics, probabilities = topic_model.transform(test_data["Sentences"],)
topics

Am I missing something? 

Thanks & BR
Aymen

**Comment by MaartenGr** [comment]
There are a number of functions that you can use for extracting the name of the topic. First, you can use `topic_model.get_topic(topic)` to get the keywords of each topic which, together, make up the name. Second, you can use `topic_model.topic_labels_` to get a mapping from number to name. Third, you can use `topic_model.get_topic_info()` to get a list of topic numbers and short labels. I think there are a couple of more but these are the ones most used.

**Comment by Aymen-lab** [comment]
Thank you for your answer. Sorry, maybe my question was not clear enough.
What I wanted to know is: When I get the prediction of a sentence, I don't want to get the topic number, but rather the topic label (The customized).

Greetings,
Aymen

**Reply by MaartenGr** [reply]
When you generate a prediction, you will get back the topic number. You can then map that topic number to the custom labels by using `topic_model.custom_labels_`. These labels are ordered based on the topic number so that they can quickly be mapped.
